packages/zowe-explorer-api/src/fs/utils/jobs.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { IJobFile } from "@zowe/zos-jobs-for-zowe-sdk";
+import { IFileSystemEntry, JobEntry, SpoolEntry } from "../types";
+import { FileType } from "vscode";
+
+export function isJobEntry(entry: IFileSystemEntry): entry is JobEntry {
+    return entry != null && entry.type == FileType.Directory;
+}
+
+export function isSpoolEntry(entry: IFileSystemEntry): entry is SpoolEntry {
```

Comment:
I'm really starting to see the benefits in using the typeguard. Not saying that we should restructure the code to prefer them over TS Generics like with the `IZoweTree` (since they may not be 1-1 replaceable due to some obscure edge case). However, maybe we should use them more often going forward to avoid unnecessary double/triple type-checks 😋